llama-index-integrations/vector_stores/llama-index-vector-stores-opensearch/llama_index/vector_stores/opensearch/base.py

```diff
@@ -123,14 +123,18 @@ def __init__(
         self._os_async_client = os_async_client or self._get_async_opensearch_client(
```

Comment:
just curious, there's no way to create an sync client from an async client, or vice-versa? This would solve a lot of headaches if possible

Reply:
I guess you could if you dig into the kwargs and change it accordingly for each sync/async instantiation
```
from opensearchpy import AsyncHttpConnection, AWSV4SignerAsyncAuth, RequestsHttpConnection, AWSV4SignerAuth

# Sync parameters
kwargs['auth'] = AWSV4SignerAuth(credentials, region, 'aoss')
kwargs['connection_class'] = RequestsHttpConnection
self._os_client = os_client or self._get_opensearch_client(
    self._endpoint, **kwargs
)

# Async parameters
kwargs['auth'] = AWSV4SignerAsyncAuth(credentials, region, 'aoss')
kwargs['connection_class'] = AsyncHttpConnection
self._os_client = os_client or self._get_opensearch_client(
    self._endpoint, **kwargs
)
```

But you'd need the AWS credentials/region/service ('aoss') as additional inputs, seems too much

Reply:
ah yea, thats pretty messy. Too bad